### DISCUSSION TITLE
Write MagickImageCollection to PDF with images of different sizes.

### DISCUSSION THREAD

**Discussion by harleygnuya** [dlemstra/Magick.NET#1622] [Help]
I have 14 MagickImage images in a MagickImageCollection that I wish to save as a multi-page PDF with each image on its own page.  This is a view from Visual Studio 2022 of the contents of the collection at the point where I am about to write the collection to a PDF file:

`<!DOCTYPE html>
<HTML>
<head>
<title>Document</title></head>
<body>

  | Name | Value | Type
-- | -- | -- | --
◢ | Images | {ImageMagick.MagickImageCollection} | ImageMagick.MagickImageCollection
  | Count | 14 | int
  | IsReadOnly | false | bool
  | ◢ Non-Public members |   |  
  | ◢ _images | Count = 14 | System.Collections.Generic.List<ImageMagick.IMagickImage<byte>>
  | ▶ [0] | {Pdf 5100x3300 8-bit sRGB} | ImageMagick.IMagickImage<byte> {ImageMagick.MagickImage}
  | ▶ [1] | {Pdf 2550x3300 8-bit sRGB} | ImageMagick.IMagickImage<byte> {ImageMagick.MagickImage}
  | ▶ [2] | {Pdf 2550x3300 8-bit sRGB} | ImageMagick.IMagickImage<byte> {ImageMagick.MagickImage}
  | ▶ [3] | {Pdf 2550x3300 8-bit sRGB} | ImageMagick.IMagickImage<byte> {ImageMagick.MagickImage}
  | ▶ [4] | {Pdf 2550x3300 8-bit sRGB} | ImageMagick.IMagickImage<byte> {ImageMagick.MagickImage}
  | ▶ [5] | {Pdf 2550x3300 8-bit sRGB} | ImageMagick.IMagickImage<byte> {ImageMagick.MagickImage}
  | ▶ [6] | {Pdf 2550x3300 8-bit sRGB} | ImageMagick.IMagickImage<byte> {ImageMagick.MagickImage}
  | ▶ [7] | {Pdf 5100x3300 8-bit sRGB} | ImageMagick.IMagickImage<byte> {ImageMagick.MagickImage}
  | ▶ [8] | {Pdf 2550x3300 8-bit sRGB} | ImageMagick.IMagickImage<byte> {ImageMagick.MagickImage}
  | ▶ [9] | {Pdf 2550x3300 8-bit sRGB} | ImageMagick.IMagickImage<byte> {ImageMagick.MagickImage}
  | ▶ [10] | {Pdf 2550x3300 8-bit sRGB} | ImageMagick.IMagickImage<byte> {ImageMagick.MagickImage}
  | ▶ [11] | {Pdf 2550x3300 8-bit sRGB} | ImageMagick.IMagickImage<byte> {ImageMagick.MagickImage}
  | ▶ [12] | {Pdf 2550x3300 8-bit sRGB} | ImageMagick.IMagickImage<byte> {ImageMagick.MagickImage}
  | ▶ [13] | {Pdf 2550x3300 8-bit sRGB} | ImageMagick.IMagickImage<byte> {ImageMagick.MagickImage}

</body>
</HTML>
`  
I apologize for the formatting, this is the first time I'm writing to this forum.

As you can see, two of the images are 5100x3300 pixels and the rest are all 2550x3300.

When I execute this statement:

`Images.Write(PageImages[1].ImageFileName.PDFPathName, MagickFormat.Pdf);`

where PageImages[1].ImageFileName.PDFPathName is the filename and MagickFormat.Pdf is the format, a good, proper and readable pdf file is generated with all of the images on their own pages and in proper sequence, but all of the images are the same size: 5100x3300 pixels.  The smaller (2550x3300) images appear in the PDF as if  they were drawn onto a 5100x3300 canvas with northwest gravity.

How can write the PDF file so that all of the images retain their original sizes in the PDF file?

.NET 8.0, C# 12.0, Magick.NET-Q8-AnyCPU(13.6.0), Magick.NET.SystemWindosMedia (7.2.2), Visual Studio 2022

**Comment by dlemstra** [comment]
That is probably how your PDF viewer works? If you would wand to have the same size you probably need to resize all the images? Or you could play around with the density of the image?

**Reply by harleygnuya** [reply]
Thanks for your reply.  I tried adjusting the density but the only effect that had was to change the dpi on the output images.  The unexpected behavior of constructing the PDF pages with the smaller images drawn onto a canvas the same size as the larger images is still happening.  It appears as if this is a matter of the way the PDF is being constructed, as the pages of the PDF appear the same whether I view them in Acrobat Pro 2024, Readiris PDF 22, or Kofax Power PDF Assistant 4.1.  Just so you know, I have Ghostscript 10.03.0 64-bit installed on my computer.

This a screenshot of a folder containing the individual tif images written individually from the MagickImageCollection:

![image](https://github.com/dlemstra/Magick.NET/assets/29216340/ee28ec9c-9fa6-464d-9db3-e67ba321a25c)

This illustrates how each image should appear.  In other words, they seem to be correct as stored in the collection.

This is the properties of one of the smaller images:

![image](https://github.com/dlemstra/Magick.NET/assets/29216340/29d346b1-e742-48e4-9e6d-db8f94626656)

The larger ones are the same with the exception their being 5100 x 3300 pixels.

This is a screenshot of the Organize Pages panel of Acrobat Pro that shows the appearance of the pages in the PDF:

![image](https://github.com/dlemstra/Magick.NET/assets/29216340/e3d83829-5630-4070-98e5-9ccd884175ac)

The page panels of the other readers show the same thing.

Notice that the two large images, pages 1 and 8, are correct.

On the rest of the odd pages, the smaller images appear as a correctly-sized image placed on the left side of a larger canvas that is the same size as the larger pages.  What's even worse is that with the even pages, all we get is an empty canvas.

Is there anything else I can try?

**Reply by dlemstra** [reply]
Your first image has a different orientation? What happens if you rotate it?

**Reply by harleygnuya** [reply]
Here's a screenshot from Acrobat.  All it did was to rotate the two larger images.  It had no effect on the others.

![image](https://github.com/dlemstra/Magick.NET/assets/29216340/fb55c1a3-aa02-4a24-90ed-1d5ea68ef177)

**Comment by dlemstra** [comment]
I haven't had time to investigate this yet. Still not clear what causes this. Do you have any updates?

**Reply by harleygnuya** [reply]
Thanks for following up.  I've been away from home the past two weeks, so I haven't been able to do any more with this issue during that time.  I will look at it again tomorrow and get back with you.

**Reply by dlemstra** [reply]
I think a solution could be to resize the landscape images (5100x3300) to the same width as the portrait images (2550x3300). In other words `image.Resize(2550,0);`
